### ISSUE TITLE
What is the new naming scheme?

### ISSUE THREAD

**Issue by idbrii** [tpope/vim-fugitive#1816]
d10dc9ea93428acc11c54aa22b985c6392ff897c / 8b0a40df quarantine deprecated commands in documentation with comment:

> The following commands are deprecated in favor of replacements that adhere to a new naming scheme.

But I can't figure out the new scheme. 
Aside from the two main (`:Git` `:G`) commands, there are a few GUppercase commands:

* `:GBrowse`
* `:GDelete`
* `:GMove`
* `:GRemove`
* `:GRename`

And many Glowercase commands:

* `:Gcd`
* `:Gclog`
* `:Gdiffsplit`
* `:Gedit`
* `:Ggrep`
* `:Ghdiffsplit`
* `:Glcd`
* `:Glgrep`
* `:Gllog`
* `:Gpedit`
* `:Gread`
* `:Gsplit`
* `:Gtabedit`
* `:Gvdiffsplit`
* `:Gvsplit`
* `:Gwq`
* `:Gwrite`

The GUppercase commands map directly to git commands (except `:GBrowse`), but
with slightly different names. Some Glowercase commands also map to git
commands (`:Ggrep`, `:Gdiffsplit`, `:Gllog`) but have variants. Most but not all GUppercase commands modify the index, but so does `:Gwrite`.

Are all the commands going to be GUppercase or is there a meaning to GUppercase that I'm missing?

**Comment by tpope** [comment]
The lowercase commands map directly to _Vim_ commands: `:Gcd` is `:cd`, `:Ggrep` is `:grep`, etc. The exceptions that prove the rule:

* `:Gclog`/`:Gllog`. These are like `:cfile`/`:lfile`, `:cexpr`/`:lexpr`, etc, but for `git log`. I regret not making these `:GcLog`/`:GlLog`, doubly so after seeing your confusion, but I don't think I wanna do a second deprecation cycle right after the first. And also, my ultimate fantasy is to get rid of these entirely.
* `:Ghdiffsplit`: This is `:Gdiffsplit` but forced horizontal. I especially hate this one but what can you do.

The uppercase commands are commands that _don't_ map directly to Vim commands. Some of them do have Vim commands with the same name—there is a `:move` for example, but it moves lines within the file, and is unrelated to what `:Gmove` does, which is why I renamed it to `:GMove`. It actually corresponds to `:Move` from [/tpope/vim-eunuch](eunuch.vim), as do a few others.

In summary, the naming scheme is `G` plus whatever the non-Git equivalent command is.

**Comment by idbrii** [comment]
Thanks for the clarification!

> I regret not making these :GcLog/:GlLog, doubly so after seeing your confusion

I must admit, I'm glad you didn't! I find MixEdCase commands with short words difficult to type and have to resort to tab completion.

**Comment by lachesis** [comment]
Yeah I really wish you hadn't renamed Gbrowse. I don't use a leader key for that one, just type `:Gbrowse!`. Apparently I have a lot of muscle memory there.

**Comment by tpope** [comment]
If it makes you feel better, I'm currently in no rush to remove `:Gbrowse`, as it's not really hurting anything other than polluting the tab complete namespace a bit. Contrast with the subcommand wrappers like `:Gcommit`, which actively confuse people and can't be removed soon enough.

You can also define `:Gbrowse` in your vimrc if you want and keep using it till the end of time. Though exactly passing along the line numbers to `:GBrowse` can be tricky.

**Comment by idbrii** [comment]
The way [fugitive_legacy_commands are defined](https://github.com/tpope/vim-fugitive/blob/404c3c24e106cf273406c0bd836e6bd4ed87e9f0/plugin/fugitive.vim#L505-L580) makes it easy to copypaste to your vimrc and strip off the echohl.

**Comment by idbrii** [comment]
@drahnieR You can always define your own commands. `:command! Gblame Git blame`

For all the features, see also:

> The way [fugitive_legacy_commands are defined](https://github.com/tpope/vim-fugitive/blob/404c3c24e106cf273406c0bd836e6bd4ed87e9f0/plugin/fugitive.vim#L505-L580) makes it easy to copypaste to your vimrc and strip off the echohl.

(But beware tpope's caveat.)
